packages/langium/src/grammar/langium-grammar-code-actions.ts

```diff
@@ -58,4 +66,76 @@ export class LangiumGrammarCodeActionProvider implements CodeActionProvider {
             }
         };
     }
-}
\ No newline at end of file
+
+    private fixRegexTokens(diagnostic: Diagnostic, document: LangiumDocument): CodeAction | undefined {
+        const offset = document.textDocument.offsetAt(diagnostic.range.start);
+        const rootCst = document.parseResult.value.$cstNode;
+        if (rootCst) {
+            const cstNode = findLeafNodeAtOffset(rootCst, offset);
+            const element = cstNode?.element;
+            const container = ast.isCharacterRange(element) ? element : getContainerOfType(element, ast.isCharacterRange);
+            if (container && container.right && container.$cstNode) {
+                const left = container.left.value;
+                const right = container.right.value;
+                return {
+                    title: 'Refactor into regex',
```

Comment:
I would write the longer term "regular expression" in such user-facing strings, unless there's a real need to shorten it.